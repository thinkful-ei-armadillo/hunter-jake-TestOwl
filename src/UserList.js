import React from "react";
import STORE from './store'

function UserList() {
  let onStage = [];
  let inSession = [];
  for(let i =0; i<STORE.participants.length; i++){
    if(STORE.participants[i].onStage){
      onStage.push(STORE.participants[i])
    } else if (STORE.participants[i].inSession){
      inSession.push(STORE.participants[i])
    }
  }
  

  return(
    <div className = 'ActiveList'>
    <ul className="onStage">
      {onStage.map(onStage =>  
      <li>
        <div className="onStageParagraph">
          <p className="stageName">{onStage.name}</p>
          <p>On Stage</p>
        </div>
        <img src={onStage.avatar} alt="onStage.name" />
        </li>)}
      </ul>

      <ul className="inSession">
      {inSession.map(inSession =>  
      <li>
        <p className="sessionName">{inSession.name}</p>
         <p>In Session</p>
         <img src={inSession.avatar} alt="inSession.name" />
         </li>)}
      </ul>
    </div>
  )
}

export default UserList;