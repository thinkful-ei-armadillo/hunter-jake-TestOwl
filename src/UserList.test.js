import React from 'react';
import ReactDOM from 'react-dom';
import UserList from './UserList';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<UserList />', () => {
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<UserList onStage="[]" inSession="[]"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
})