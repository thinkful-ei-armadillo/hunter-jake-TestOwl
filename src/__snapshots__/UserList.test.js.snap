// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> renders the UI as expected 1`] = `
<div
  className="ActiveList"
>
  <ul>
    <li>
      <img
        alt="onStage.name"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <p>
        Koren Templeton
      </p>
      <p>
        On Stage
      </p>
    </li>
    <li>
      <img
        alt="onStage.name"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <p>
        Ashla Attwool
      </p>
      <p>
        On Stage
      </p>
    </li>
  </ul>
  <ul>
    <li>
      <img
        alt="inSession.name"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <p>
        Caty Flucker
      </p>
      <p>
        In Session
      </p>
    </li>
    <li>
      <img
        alt="inSession.name"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <p>
        Frank Runciman
      </p>
      <p>
        In Session
      </p>
    </li>
  </ul>
</div>
`;
